kafkakewl-metrics/src/main/scala/com/mwam/kafkakewl/metrics/services/ConsumerOffsetsSource.scala

```diff
@@ -191,17 +195,24 @@ object ConsumerOffsetsSource {
         Deserializer.bytes,
         Deserializer.bytes
       )
-      .map { cr =>
+      .mapZIO { cr =>
         val topic = cr.record.topic
-        val key = ConsumerOffsetsDeserializers.keyDeserializer.deserialize(topic, cr.key.get)
+        val key: IO[Throwable, Option[ConsumerGroupTopicPartition]] =
+          ConsumerOffsetsDeserializers.keyDeserializer.deserialize(topic, new RecordHeaders(), cr.key.get)
         // TODO refactor this to re-use elsewhere
         // the key deserializer can return null in which case we need to skip this key-value altogether
-        if (key == null) {
-          None
-        } else {
-          val value = Option(cr.value).map(valueBytes => ConsumerOffsetsDeserializers.valueDeserializer.deserialize(topic, valueBytes.get))
-          Some((key, value))
-        }
+        val value: IO[Throwable, Option[KafkaConsumerGroupOffset]] =
+          Option(cr.value) match {
+            case Some(value) => ConsumerOffsetsDeserializers.valueDeserializer.deserialize(topic, new RecordHeaders(), value.get).map(Some(_))
+            case None        => ZIO.succeed(None)
+          }
+
+        val zipped: IO[Throwable, Option[(ConsumerGroupTopicPartition, Option[KafkaConsumerGroupOffset])]] =
+          key
+            .zipWith(value)((key, value) => key.zip(Some(value)))
+            .orElseFail(new Throwable("fail"))
```

Comment:
Not sure if this orElseFail is really needed - the error type is already throwable?